hatch_zipped_directory/builder.py

```diff
@@ -24,26 +26,43 @@
 
 __all__ = ["ZippedDirectoryBuilder"]
 
+ZipTime = Tuple[int, int, int, int, int, int]
+
 
 class ZipArchive:
-    def __init__(self, zipfd: ZipFile, root_path: str):
+    def __init__(self, zipfd: ZipFile, root_path: str, *, reproducible: bool):
         self.root_path = PurePosixPath(root_path)
         self.zipfd = zipfd
+        self.reproducible = reproducible
+        self.timestamp: int | None = (
+            get_reproducible_timestamp() if reproducible else None
+        )
+        self.ziptime: ZipTime | None = (
+            time.gmtime(self.timestamp)[0:6] if self.timestamp else None
+        )
 
     def add_file(self, included_file: IncludedFile) -> None:
         arcname = self.root_path / included_file.distribution_path
-        self.zipfd.write(included_file.path, arcname=arcname)
+        info = ZipInfo.from_file(included_file.path, arcname)
+        if self.ziptime:
+            info.date_time = self.ziptime
+        with open(included_file.path, "rb") as f:
+            self.zipfd.writestr(info, f.read())
```

Comment:
I honestly did not consider large files, kicad plugins are size limited so that's not real concern for me. But I agree that hatch-ziped-directory should use better approach.